entgql/template/collection.tmpl

```diff
@@ -166,8 +166,13 @@ func {{ $newPaginateArg }}(rv map[string]interface{}) *{{ $paginateArg }} {
 		{{- $withWhere := true }}{{ with $node.Annotations.EntGQL }}{{ if isSkipMode .Skip "where_input" }}{{ $withWhere = false }}{{ end }}{{ end }}
 		{{- if $withWhere }}
 			{{- $where := $names.WhereInput }}
-			if v := rv[whereField]; v != nil && v != (*{{ $where }})(nil) {
-				args.opts = append(args.opts, {{ $filter }}(v.(*{{ $where }}).Filter))
+			if v, ok := rv[whereField]; ok {
+				switch v := v.(type) {
```

Comment:
Why not if?

Reply:
Just make sure it doesn't panic in any case. 🙃

Reply:
Ah, Got it.